lib/view_model/pre_auth_view_models/login_view_model.dart

```diff
@@ -60,7 +77,29 @@ class LoginViewModel extends BaseModel {
     ];
   }
 
-  /// This function is used to sign-in the user into application.
+  /// This function performs the login operation.
+  ///
+  /// **params**:
+  ///   None
+  ///
+  /// **returns**:
+  /// * `Future<void>`: a promise that indicates the completion of the login process.
+  ///
+  /// This function handles the login process by performing the following steps:
+  /// 1. Unfocusing the email and password text fields.
+  /// 2. Setting validation mode to `AutovalidateMode.always`.
+  /// 3. Validating the email and password fields using the form key.
+  /// 4. If validation is successful, disabling auto-validation mode and initiating the login process.
+  /// 5. Displaying a custom progress dialog during login.
+  /// 6. Initializing database functions.
+  /// 7. Performing a GraphQL mutation to login the user by providing the email and encrypted password.
+  /// 8. Handling the result of the login operation:
+  ///    - Updating the current user with the received data.
+  ///    - Redirecting the user based on their status in the application.
+  ///    - Handling Firebase options for Android and iOS if available.
+  ///    - Configuring Firebase and saving FCM token to the database.
+  ///
+  /// In case of any exceptions during the login process, this function catches and prints the error.
   Future<void> login() async {
```

Comment:
Very good documentation.

Reply:
Thank you very much for your appreciation. I'm grateful for your recognition of my work.

Reply:
Hey @Azad99-9, I updated the comment during my previous review but somehow it didn't. Please check the previous comment again